app/src/components/overlay/Modal.tsx

```diff
@@ -101,6 +101,12 @@ const modalCSS = css`
       left: 50%;
       transform: translate(-50%, -50%);
       z-index: 1001;
+      // 90% gives a decent amount of padding around the dialog when it would
+      // otherwise be cut off by the edges of the screen
+      max-height: 90%;
```

Comment:
maybe prefer calc(100% - gap) that way things stay a bit more well organized in terms of spacing.